pkg/venafi/cloud/connector.go

```diff
@@ -827,6 +987,96 @@ func (c *Connector) RetrieveSystemVersion() (response string, err error) {
+	var cloudOptions *cloudkeystores.CertificateProvisioningOptionsInput
+	dataOptions, err := json.Marshal(options)
+	if err != nil {
+		return nil, err
+	}
+
+	graphqlAzureOptions := &cloudkeystores.CertificateProvisioningAzureOptionsInput{}
+	graphqlGCPOptions := &cloudkeystores.CertificateProvisioningGCPOptionsInput{}
+
+	if options != nil {
+		switch (*options).GetType() {
+		case string(cloudkeystores.CloudKeystoreTypeAcm):
+			// nothing
+		case string(cloudkeystores.CloudKeystoreTypeAkv):
+			err = json.Unmarshal(dataOptions, graphqlAzureOptions)
+			if err != nil {
+				return nil, err
+			}
+		case string(cloudkeystores.CloudKeystoreTypeGcm):
+			err = json.Unmarshal(dataOptions, graphqlGCPOptions)
+			if err != nil {
+				return nil, err
+			}
+		default:
+			return nil, fmt.Errorf("unknown cloud keystore type: %s", (*options).GetType())
+		}
+	}
+
+	cloudOptions = &cloudkeystores.CertificateProvisioningOptionsInput{
+		AwsOptions:   nil,
+		AzureOptions: graphqlAzureOptions,
+		GcpOptions:   graphqlGCPOptions,
+	}
+	return cloudOptions, nil
+}
+
+func getCloudMetadataFromWebsocketResponse(respMap interface{}, keystoreType cloudkeystores.CloudKeystoreType, keystoreId string) (*CloudProvisioningMetadata, error) {
+
+	val := CloudKeystoreProvisioningResult{}
+	valJs, err := json.Marshal(respMap)
+	if err != nil {
+		return nil, fmt.Errorf("unable to encode response data! Error: %s", err.Error())
+	}
+	err = json.Unmarshal(valJs, &val)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse response data! Error: %s", err.Error())
+	}
+	if val.Error != nil {
+		return nil, fmt.Errorf("unable to provision certificate! Error: %s", val.Error)
+	}
+
+	if val.CloudProviderCertificateId == "" {
+		return nil, fmt.Errorf("provisioning is not successful, certificate ID from response is empty")
+	}
+
+	cloudMetadata := &CloudProvisioningMetadata{}
+	switch keystoreType {
+	case cloudkeystores.CloudKeystoreTypeAcm:
+		cloudMetadata.awsMetadata.result = val
+	case cloudkeystores.CloudKeystoreTypeAkv:
+		cloudMetadata.azureMetadata.result = val
+	case cloudkeystores.CloudKeystoreTypeGcm:
+		cloudMetadata.gcpMetadata.result = val
+	default:
+		err = fmt.Errorf("Unknown ConnectorType %v found for keystore with ID: %s", keystoreType, keystoreId)
+		return nil, err
+	}
+	return cloudMetadata, err
+}
+
+func validateProvisioningInputs(req endpoint.ProvisioningRequest) error {
+	if req.KeystoreId != nil && (req.KeystoreName != nil || req.ProviderName != nil) {
+		return fmt.Errorf("only should keystoreId OR keystoreName and providerName can be provided")
+	}
+	if req.KeystoreId == nil {
+		if req.KeystoreName == nil {
+			return fmt.Errorf("missing providerName")
+		}
+		if req.ProviderName == nil {
+			return fmt.Errorf("missing keystoreName")
+		}
+		return fmt.Errorf("missing to provide any of keystoreId or keystoreName and providerName")
+	}
+	if (req.CertificateId == nil && req.PickupId == nil) || (req.CertificateId != nil && req.PickupId != nil) {
```

Comment:
The same happens here, set a priority and check you have at least one of them. Do not fail by having both (pickupID and certificateID) set

Reply:
Resolved by latest commit